File: src/TubeStatusFetcher.UI.Swift/TubeStatusFetcher.UI.Swift/Model/LineInfo.swift
//
//  LineInfo.swift
//  TubeStatusFetcher.UI.Swift
//
//  Created by VOLKAN PAKSOY on 07/11/2015.
//  Copyright © 2015 VOLKAN PAKSOY. All rights reserved.
//

import UIKit

struct LineInfo {

    var Id : String
    var Name : String
    var LineColour : RGB
    var StatusSeverity : Int
    var StatusSeverityDescription : String
    var Reason : String
    
    
    init(status: AnyObject) {

        Id = status["id"] as! String
        Name = status["name"] as! String
        StatusSeverity = status["lineStatuses"]!![0]!["statusSeverity"] as! Int
        StatusSeverityDescription = status["lineStatuses"]!![0]!["statusSeverityDescription"] as! String
        Reason = ""
        
        LineColour = RGB(R: 0, G: 0, B: 0)
        
        
    }
    
    
    static func lineStatusFromResults(results: [AnyObject]) -> [LineInfo] {
        var lineStatus = [LineInfo]()
        
        for result in results {
            
            lineStatus.append(LineInfo(status: result))
        }
        
        return lineStatus
    }
    
}

File: src/TubeStatusFetcher.UI.Xamarin/Core/LineInfo.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TubeStatusFetcher.Core
{
    public class LineInfo
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public RGB LineColour { get; set; }
        public int StatusSeverity { get; set; }
        public string StatusSeverityDescription { get; set; }
        public string Reason { get; set; }
    }
}
